src/Compilers/CSharp/Portable/CSharpParseOptions.cs

```diff
@@ -228,6 +173,32 @@ public new CSharpParseOptions WithFeatures(IEnumerable<KeyValuePair<string, stri
             }
         }
 
+        internal override void ValidateOptions(ArrayBuilder<Diagnostic> builder)
+        {
+            ValidateOptions(builder, MessageProvider.Instance);
+
+            // Validate LanguageVersion not SpecifiedLanguageVersion, after Latest/Default has been converted:
+            if (!LanguageVersion.IsValid())
+            {
+                builder.Add(Diagnostic.Create(MessageProvider.Instance, (int)ErrorCode.ERR_BadLanguageVersion, LanguageVersion.ToString()));
+            }
+            
+            if (!PreprocessorSymbols.IsDefaultOrEmpty)
+            {
+                foreach (var symbol in PreprocessorSymbols)
+                {
+                    if (symbol == null)
+                    {
+                        builder.Add(Diagnostic.Create(MessageProvider.Instance, (int)ErrorCode.WRN_DefineIdentifierRequired, "null"));
```

Comment:
I chose it to match the warning returned by the command line parser. I believe they should match. The user shouldn't get a warning from the command line and an error from the API (for the same input).
Should we turn both into an error? thoughts?

---
In reply to: [105716480](https://github.com/dotnet/roslyn/pull/17113#discussion_r105716480) [](ancestors = 105716480)